save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/topbar/TopBarLinks.kt

```diff
@@ -0,0 +1,127 @@
+    var location: Location
+}
+
+/**
+ * @property hrefAnchor is link
+ * @property width is width of the link text
+ * @property text is link text
+ */
+data class TopBarLink(
+    val hrefAnchor: String,
+    val width: Width,
+    val text: String,
+)
+
+/**
+ * Displays the static links that do not depend on the url
+ */
+@Suppress("MAGIC_NUMBER", "LongMethod", "TOO_LONG_FUNCTION")
+private fun topBarLinks() = FC<TopBarLinksProps> { props ->
+    val navigate = useNavigate()
+    val (isDemoDropdownActive, setIsDemoDropdownActive) = useState(false)
+
+    ul {
+        className = ClassName("navbar-nav mx-auto")
+        li {
+            className = ClassName("nav-item dropdown no-arrow")
+            style = jso {
+                width = 5.rem
+            }
+            a {
+                className = ClassName("nav-link dropdown-toggle text-light")
+                asDynamic()["data-toggle"] = "dropdown"
+                ariaExpanded = false
+                id = "demoDropdown"
+                role = "button".unsafeCast<AriaRole>()
+                +"Demo"
+                onClickCapture = { _ ->
+                    setIsDemoDropdownActive { !it }
+                }
+            }
+            div {
+                className = ClassName("dropdown-menu dropdown-menu-right shadow animated--grow-in${if (isDemoDropdownActive) " show" else "" }")
+                ariaLabelledBy = "demoDropdown"
+                val diktatDemoHref = "/${FrontendRoutes.DEMO.path}/diktat"
+                dropdownEntry(null, "Diktat", window.location.href.contains(diktatDemoHref)) { attrs ->
+                    attrs.onClick = {
+                        setIsDemoDropdownActive(false)
+                        navigate(to = diktatDemoHref)
+                    }
+                }
+                val cpgDemoHref = "/${FrontendRoutes.DEMO.path}/cpg"
+                dropdownEntry(null, "CPG", window.location.href.contains(cpgDemoHref)) { attrs ->
+                    attrs.onClick = {
+                        setIsDemoDropdownActive(false)
+                        navigate(to = cpgDemoHref)
+                    }
+                }
+            }
+        }
+        listOf(
+            TopBarLink(hrefAnchor = FrontendRoutes.AWESOME_BENCHMARKS.path, width = 12.rem, text = "Awesome Benchmarks"),
+            TopBarLink(hrefAnchor = FrontendRoutes.SANDBOX.path, width = 9.rem, text = "Try SAVE format"),
+            TopBarLink(hrefAnchor = SAVE_CLOUD_GITHUB, width = 9.rem, text = "SAVE on GitHub"),
+            TopBarLink(hrefAnchor = FrontendRoutes.PROJECTS.path, width = 8.rem, text = "Projects board"),
+            TopBarLink(hrefAnchor = FrontendRoutes.CONTESTS.path, width = 6.rem, text = "Contests"),
+            TopBarLink(hrefAnchor = FrontendRoutes.ABOUT_US.path, width = 6.rem, text = "About us"),
+        ).forEach { elem ->
+            li {
+                className = ClassName("nav-item")
+                if (elem.hrefAnchor != SAVE_CLOUD_GITHUB) {
```

Comment:
Please add an extra boolean flag (e.g.: `isExternalLink: Boolean`) to `TopBarLink` instead.

@nulls

Reply:
Done [faec44c](https://github.com/saveourtool/save-cloud/pull/1540/commits/faec44ca7ee13b63b5e76ba2be295039cdb7c3e1)